_pytest/test_sanity.py

```diff
@@ -0,0 +1,66 @@
+import importlib
+import pkgutil
+
+import pytest
+from six import PY2
+
+
+def import_submodules(package):
+    """ Import all submodules of a module, recursively, including subpackages."""
+
+    if isinstance(package, str):
+        package = importlib.import_module(package)
+    results = []
+    for loader, name, is_pkg in pkgutil.walk_packages(package.__path__):
+        full_name = package.__name__ + '.' + name
+        module = importlib.import_module(full_name)
+        if PY2:
+            reload(module)
+        else:
+            importlib.reload(module)
+        results.append(full_name)
+        if is_pkg:
+            results += import_submodules(full_name)
+    return results
+
+
+@pytest.mark.parametrize("banned_package", ["spacy", "mitie", "numpy", "sklearn"])
+def test_no_global_imports_of_banned_package(banned_package):
+    """This test ensures that neither of the banned packages are imported module wise in any of our code files.
```

Comment:
They are optional dependencies, e.g. users shouldn't be required to install mitie if they only use spacy components.